Submission of snippets

Akorax opened the pull request:
Snippety ship.

Akorax committed:
Added when/if

Just testing the function of proposing changes.

cspotcode commented on .vscode/manic-miners.code-snippets:
```diff
@@ -3,16 +3,28 @@
-}
\ No newline at end of file
+	"When": {
+		"scope": "javascript,typescript",
```
As discussed on Discord, we'll need to remove these "scope" entries.  Eventually we can have a "manicminerslevel" language which can go here.

I know you already know this, but putting a comment shows how we can discuss changes on github.

Akorax committed:
First actual submit

Added Variables, Triggers, Events and Player interaction events.
TODO:
INIT event
Macros (except random)
Arrow Events
Unit Events
Event Chains
Conditions and Branches (there are some code but it's a test only)
Objectives doesn't work ingame I think.
Disable/Enable events
Math Events

cspotcode commented on .vscode/manic-miners.code-snippets:
```diff
@@ -15,4 +15,225 @@
+	"If": {
+		"prefix": "if()",
+		"body": "if(${1:Trigger})[${2:Event}]",
+		"description": "Writes If and put the cursor inside the parenthesis."
```
Should we update these descriptions to focus on the meaning of `When` and `If`, to help the scripter understand what they're doing?

Akorax replied:
I think very few will look at the descriptions. If they do they probably do it on the wiki where the most updated info is.

cspotcode replied:
That's fair, but these descriptions are still inconsistent with the others.  This one describes what is being typed, whereas the others focus on what the code means in the game.

cspotcode commented on .vscode/manic-miners.code-snippets:
```diff
@@ -15,4 +15,225 @@
+	},
+	"Time": {
+		"prefix": "TriggerTime",
+		"body": "time:${1:row},${2:column}",
```
update placeholders to get rid of row, column.

cspotcode reviewed (changes requested):
I think I found a typo.  Also made a suggestion about some descriptions.  I technically didn't look at *all* of it, but this looks like a great start.

cspotcode commented:
These changes have been incorporated via https://github.com/cspotcode/ManicMinersScripting/blob/master/vscode-extension/generate-code-snippets.js
